app/models/status.rb

```diff
@@ -335,6 +335,17 @@ def in_chosen_languages(account)
       where(language: nil).or where(language: account.chosen_languages)
     end
 
+    def as_home_timeline(account, limit = 20, max_id = nil, since_id = nil, min_id = nil)
+      selected = where(account: [account] + account.following)
+                 .where(visibility: [:public, :unlisted, :private])
```

Comment:
If this inlined at home_feed.rb:from_database, It could be resolved by removing `.where(visibility: [:public, :unlisted, :private])` clause and make it filtered on latter `.reject { |status| FeedManager.instance.filter?(:home, status, account) }`
How about your opinion?

Reply:
that would return less statuses than requested